### ISSUE TITLE
$expand operation not respecting the `filter` parameter

### ISSUE THREAD

**Issue by qwerty0981** [hapifhir/hapi-fhir#2004]
**Describe the bug**
When we perform a `$expand` operation on our FHIR server on the US Core Condition Code ValueSet, found [here](https://build.fhir.org/ig/HL7/US-Core-R4/ValueSet-us-core-condition-code.html), the server ignores the `filter` parameter. This seems to only happen if we initiate the `$expand` operation using the `/ValueSet/$expand?filter=fracture&url=http://hl7.org/fhir/us/core/ValueSet/us-core-condition-code` format. If we make the request to `/ValueSet/us-core-condition-code/$expand?filter=fracture` the `filter` parameter is respected. This behavior seems to happen for both normal value sets and expanded value sets. We have confirmed that this behavior can be reproduced using the `service-category`, found [here](https://www.hl7.org/fhir/valueset-service-category.html).

**To Reproduce**
1. Start the HAPI FHIR Starter project
2. `PUT` the `service-category` CodeSystem, found [here](https://www.hl7.org/fhir/codesystem-service-category.html)
```
curl -X PUT -H "Content-Type: application/json" http://localhost:8080/hapi-fhir-jpaserver/fhir/CodeSystem/service-category -d @{codeSystemFile}
```
3.  `PUT` the `service-category` ValueSet, found [here](https://www.hl7.org/fhir/valueset-service-category.html)
```
curl -X PUT -H "Content-Type: application/json" http://localhost:8080/hapi-fhir-jpaserver/fhir/ValueSet/service-category -d @{valueSetFile}
```
4. Wait for the ValueSet to be expanded
5. Make a request to `http://localhost:8080/hapi-fhir-jpaserver/fhir/ValueSet/service-category/$expand?filter=Hos`
6. Make a request to `http://localhost:8080/hapi-fhir-jpaserver/fhir/ValueSet/$expand?filter=Hos&url=http://hl7.org/fhir/ValueSet/service-category`

**Expected behavior**
I expected both queries to respect the `filter` parameter and return an expansion with one element:
![image](https://user-images.githubusercontent.com/5105031/88610363-6c3d2c00-d054-11ea-9c0c-74d3bd72b312.png)
Instead the request with the ValueSet id explicitly provided respects the `filter` parameter while the query using the `url` parameter does not.

**Screenshots**
Using the ValueSet id:
![image](https://user-images.githubusercontent.com/5105031/88610486-ce962c80-d054-11ea-95ec-9255bbcf12fc.png)

Using `url` parameter:
![image](https://user-images.githubusercontent.com/5105031/88610444-b7573f00-d054-11ea-8524-7bc2ea23c083.png)

Request Logs:
![image](https://user-images.githubusercontent.com/5105031/88610801-94795a80-d055-11ea-9021-c7aa68f05999.png)

**Environment:**
 - HAPI FHIR Version: 5.0.2
 - OS: Ubuntu 18.04
 - Using built-in h2 backend for reproduction but it happens in our Postgresql backed environment as well

Please let me know if there is any additional information that you would need to diagnose the issue.

**Comment by bratwurzt** [comment]
Duplicate of #2189

**Comment by jamesagnew** [comment]
Closing, as #2189 is fixed. Please feel free to comment if this is not resolved.

**Comment by bratwurzt** [comment]
I just checked with master branch build and this is unfortunately still not resolved. Weirder still, calling expand with ValueSet id now returns error
/ValueSet/administrative-gender/$expand?filter=fema
`{
    "resourceType": "OperationOutcome",
    "issue": [
        {
            "severity": "error",
            "code": "processing",
            "diagnostics": "Resource ValueSet/administrative-gender is not known"
        }
    ]
}`
while calling
/ValueSet/$expand?url=http://hl7.org/fhir/ValueSet/administrative-gender&filter=fema
returns:
`{
    "resourceType": "ValueSet",
    "status": "active",
    "compose": {
        "include": [
            {
                "system": "http://hl7.org/fhir/administrative-gender"
            }
        ]
    },
    "expansion": {
        "identifier": "f3cd5f01-2731-4013-9695-0d94ff30eda0",
        "timestamp": "2021-02-16T21:57:01+01:00",
        "offset": 0,
        "parameter": [
            {
                "name": "offset",
                "valueInteger": 0
            },
            {
                "name": "count",
                "valueInteger": 1000
            }
        ],
        "contains": [
            {
                "system": "http://hl7.org/fhir/administrative-gender",
                "code": "male",
                "display": "Male"
            },
            {
                "system": "http://hl7.org/fhir/administrative-gender",
                "code": "female",
                "display": "Female"
            },
            {
                "system": "http://hl7.org/fhir/administrative-gender",
                "code": "other",
                "display": "Other"
            },
            {
                "system": "http://hl7.org/fhir/administrative-gender",
                "code": "unknown",
                "display": "Unknown"
            }
        ]
    }
}`

**Comment by bratwurzt** [comment]
This is still not resolved in 5.4.0. Here's a test that reproduces the issue:
```	/**
	 * A valueset expand with url and filter
	 */
	@Test
	public void testExpandValueSetContainingSystemIncludeWithCode() throws IOException {
		CodeSystem cs = loadResourceFromClasspath(CodeSystem.class, "/r4/iar/CodeSystem-iar-citizenship-status.xml");
		myCodeSystemDao.create(cs);

		ValueSet vs = loadResourceFromClasspath(ValueSet.class, "/r4/iar/ValueSet-iar-citizenship-status.xml");
		myValueSetDao.create(vs);

		ValueSet expansion = myValueSetDao.expandByIdentifier(
			"http://ccim.on.ca/fhir/iar/ValueSet/iar-citizenship-status",
			ValueSetExpansionOptions.forOffsetAndCount(0, 1000).setFilter("fugee"));
		ourLog.info(myFhirCtx.newJsonParser().setPrettyPrint(true).encodeResourceToString(expansion));

		assertEquals("Refugee", expansion.getExpansion().getContains().iterator().next().getDisplay());
	}
```
